contracts/WildcardSteward_v2.sol

```diff
@@ -168,33 +172,37 @@ contract WildcardSteward_v2 is Initializable {
         }
     }
 
-    function addTokenGenerationRateToExistingTokens(
+    function fixTotalPatronTokenGenerationRate(
         uint256[] memory tokens,
-        uint256[] memory _tokenGenerationRate
-    ) internal {
-        assert(tokens.length == _tokenGenerationRate.length);
+        address[] memory tokenPatrons
+    ) public onlyAdmin {
         for (uint8 i = 0; i < tokens.length; ++i) {
-            assert(tokenGenerationRate[tokens[i]] == 0);
+            uint256 tokenId = tokens[i];
+            address currentOwner = currentPatron[tokenId];
 
-            tokenGenerationRate[tokens[i]] = _tokenGenerationRate[i];
-        }
-    }
+            // NOTE: for this upgrade we make sure no tokens are foreclosed, or close to foreclosing
+            _collectPatronage(tokenId);
 
-    function setMintManager(address _mintManager) public {
-        require(
-            address(mintManager) == address(0),
-            "Only set on initialisation"
-        ); // This can only be called once!
-        mintManager = MintManager_v2(_mintManager);
-    }
+            uint256 timeSinceLastMint = now.sub(timeLastCollected[tokenId]);
 
-    function updateToV2(
-        address _mintManager,
-        uint256[] memory tokens,
-        uint256[] memory _tokenGenerationRate
-    ) public {
-        setMintManager(_mintManager);
-        addTokenGenerationRateToExistingTokens(tokens, _tokenGenerationRate);
+            mintManager.tokenMint(
+                currentOwner,
+                timeSinceLastMint,
+                tokenGenerationRate[tokenId]
+            );
+            emit CollectLoyalty(tokenId, currentOwner, timeSinceLastMint);
+
+            totalPatronTokenGenerationRate[currentOwner] = totalPatronTokenGenerationRate[currentOwner]
+                .add(11574074074074);
```

Comment:
Can you please put in a comment in the function how exactly this number is calculated. I'm aware its to do with 10**18 per day.

Reply:
Obviously we have the mapping `mapping(uint256 => uint256) public tokenGenerationRate;`
but these are all the same for now. So guessing thats why it is hard coded. The name of this function suggests its going to be a once off use function, but otherwise wouldn't hardcode this value for future

Reply:
Yes, exactly, I'm just saving on gas by hard coding it. I'll add a comment ASAP.